### PR TITLE
Aggregator for consensus capability

### PR DESCRIPTION
**Pull request by bolekk**

### CODE REVIEW

File: core/capabilities/consensus/types/aggregator.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import (
+	ocrcommon "github.com/smartcontractkit/libocr/commontypes"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/values"
+)
+
+type Aggregator interface {
+	// Called by the Outcome() phase of OCR reporting.
+	// The inner array of observations corresponds to elements listed in "inputs.observations" section.
+	Aggregate(previousOutcome *AggregationOutcome, observations map[ocrcommon.OracleID][]values.Value) (*AggregationOutcome, error)
```
Comment (cedric-cordenier) [inline core/capabilities/consensus/types/aggregator.go:12]: @bolekk How will the encoder know the underlying structure of the encodableOutcome (how to decode it)? I think that needs to be a Value for this to work.

Comment (bolekk) [reply core/capabilities/consensus/types/aggregator.go:12]: True, thanks! I think I can easily use a different proto type inside my AggregationOutcome.

File: core/capabilities/consensus/datafeeds/feeds_aggregator.go [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+		return nil, err
+	}
+	return &dataFeedsAggregator{
+		config:       parsedConfig,
+		mercuryCodec: mercuryCodec,
+		lggr:         lggr,
+	}, nil
+}
+
+func parseConfig(config values.Map) (DataFeedsAggregatorConfig, error) {
```
Comment (archseer) [inline core/capabilities/consensus/datafeeds/feeds_aggregator.go:118]: Note: probably easiest to do this via https://pkg.go.dev/github.com/mitchellh/mapstructure, we already use it in the pipeline tasks

**Comment by bolekk** [comment]
Moving this to common. Please review https://github.com/smartcontractkit/chainlink-common/pull/331

**Pull request closed without merging**